mlflow/artifacts/__init__.py

```diff
@@ -57,3 +59,108 @@ def download_artifacts(
     artifact_repo = get_artifact_repository(artifact_uri)
     artifact_location = artifact_repo.download_artifacts(artifact_path, dst_path=dst_path)
     return artifact_location
+
+
+def load_text(artifact_uri: str) -> str:
+    """
+    Loads the artifact contents as a string.
+
+    :param artifact_uri: artifact location
+    :return: str
+
+    .. code-block:: python
+        :caption: Example
+
+        import mlflow
+        with mlflow.start_run() as run:
+            artifact_uri = run.info.artifact_uri
+            mlflow.log_text("This is a sentence", "file.txt")
+            file_content = mlflow.artifacts.load_text(artifact_uri + "/file.txt")
+            print(file_content)
+
+    .. code-block:: text
+        :caption: Output
+
+        This is a sentence
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
+        local_artifact = download_artifacts(artifact_uri, dst_path=tmpdir)
+        with open(local_artifact) as local_artifact_fd:
+            try:
+                return str(local_artifact_fd.read())
+            except Exception:
+                raise MlflowException("Unable to form a str object from file content", BAD_REQUEST)
+
+
+def load_dict(artifact_uri: str) -> dict:
+    """
+    Loads the artifact contents as a dictionary.
+
+    :param artifact_uri: artifact location
+    :return: dict
+
+    .. code-block:: python
+        :caption: Example
+
+        import mlflow
+        with mlflow.start_run() as run:
+            artifact_uri = run.info.artifact_uri
+            mlflow.log_dict({"mlflow-version": "0.28", "n_cores": "10"}, "config.json")
+            config_json = mlflow.artifacts.load_dict(artifact_uri + "/config.json")
+            print(config_json)
+
+    .. code-block:: text
+        :caption: Output
+
+        {'mlflow-version': '0.28', 'n_cores': '10'}
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
+        local_artifact = download_artifacts(artifact_uri, dst_path=tmpdir)
+        with open(local_artifact) as local_artifact_fd:
+            try:
+                return json.load(local_artifact_fd)
+            except json.JSONDecodeError:
+                raise MlflowException("Unable to form a JSON object from file content", BAD_REQUEST)
+
+
+def load_image(artifact_uri: str) -> "Image.Image":
```

Comment:
I think this typehint is causing a docstring error:

```
/home/circleci/project/mlflow/artifacts/__init__.py:docstring of mlflow.artifacts.load_image:: WARNING: py:class reference target not found: Image.Image
```

I think it's acceptable to remove the type hint for now.

Reply:
@dbczumar Sorry about missing CI failures. I have removed the type hint. Will look into this in detail and create enhancement PR.

Reply:
@dbczumar done :)